### DISCUSSION TITLE
How to find out if a PVC is "terminating"?

### DISCUSSION THREAD

**Discussion by rabejens** [kubernetes-client/csharp#1104] [Q&A]
When I have a PVC which is "terminating", when I do a `ReadNamespacedPersistentVolumeClaimAsync` or `...VolumeClaimStatusAsync` (What is the difference by the way?), I still get `Bound` as its status phase.

This is also the case when I manually `kubectl get pvc/myclaim -o yaml`.

I only see `Terminating` when I `describe` the PVC.

Is it somehow possible to get this information with the C# client?

**Comment by rabejens** [comment]
I found a workaround: If something is scheduled for deletion but stuck in `Terminating`, it has a deletion timestamp, so you can do something like:

```csharp
var pvc = await kube.ReadNamespacedPersistentVolumeClaimAsync("some-claim", "mynamespace");
if(pvc.Metadata.DeletionTimestamp.HasValue) {
  // Do something when deletion is scheduled
}
```

**Reply by tg123** [reply]
https://github.com/kubernetes-client/csharp/discussions/1139
